Add development chain-spec file for minimal/parachain templates for Omni Node compatibility (#6529)
paritytech/polkadot-sdk: templates/minimal/README.md

- 🤏 This template is a minimal (in terms of complexity and the number of components)
template for building a blockchain node.

> 

Reviewer: nit:
```suggestion
```

- 🔧 Its runtime is configured with a single custom pallet as a starting point, and a handful of ready-made pallets
such as a [Balances pallet](https://paritytech.github.io/polkadot-sdk/master/pallet_balances/index.html).

> 

Reviewer: nit:
```suggestion
```

- 🤏 This template is a minimal (in terms of complexity and the number of components)
template for building a blockchain node.

- 🔧 Its runtime is configured with a single custom pallet as a starting point, and a handful of ready-made pallets
such as a [Balances pallet](https://paritytech.github.io/polkadot-sdk/master/pallet_balances/index.html).

- 👤 The template has no consensus configured - it is best for experimenting with a single node network.

> - For the most up-to-date chain specification, refer to [staging-chain-spec-builder](https://crates.io/crates/staging-chain-spec-builder)

Reviewer: I would remove this note from the intro. It isn't very relevant from an introduction standpoint for the template, and the latest info for the chain spec isn't necessarily part of `staging-chain-spec-builder`. I would keep the `#Intro` section as before.

You don't need to add a mention about `staging-chain-spec-builder` in other place either. We already have it mentioned in the steps below (e.g. under `Starting a Minimal Template Chain`/`Omni Node`/`Install staging-chain-spec-builder`.

- 🤏 This template is a minimal (in terms of complexity and the number of components)
template for building a blockchain node.

> - 🔧 To simplify the process of using Zombienet with Omni Node, we've added a pre-configured development
> chain spec (dev_chain_spec.json) to the minimal template. Additionally, the zombienet-omni-node.toml file
> has been updated to include a valid chain spec path pointing to dev_chain_spec.json

Reviewer: I would move this to the `#### Update `zombienet-omni-node.toml` with a valid chain spec path` section, right before the network spec sample where we instruct where the chain_spec_path updating should be done.

```suggestion
- 🔧 To simplify the process of starting the minimal-template with Zombienet and Omni Node, we've added a pre-configured development
chain spec (dev_chain_spec.json) to the minimal template. The `zombienet-omni-node.toml` file of this template points to it, but you can update it to an updated chain spec generated on your machine, as shown below:
```

Reviewer: Unresolving this. Suggested to also move this to a more appropriate section. LMK what you think.

Author: I have made the appropriate changes @iulianbarbu.  FYI creating the script gh-issue

Author: https://github.com/paritytech/polkadot-sdk/issues/7267

Author: I guess you could assign it to me if you don't mind 😀

- 🔧 Its runtime is configured with a single custom pallet as a starting point, and a handful of ready-made pallets
such as a [Balances pallet](https://paritytech.github.io/polkadot-sdk/master/pallet_balances/index.html).

- 👤 The template has no consensus configured - it is best for experimenting with a single node network.

For the most accurate chain spec generation,  `staging-chain-spec-builder` or other mentioned binaries and runtimes,
please try to build them based on the Polkadot-sdk most updated master branch an example:
  ```bash
  cargo install --path substrate/bin/utils/chain-spec-builder

>   ```

Reviewer: This is a good note, but I wouldn't add it anymore. Latest `staging-chain-spec-builder` is good enough for what this readme needs: https://crates.io/crates/staging-chain-spec-builder

Reviewer 2: We also need `--locked` here. Please doublecheck all the install commands. `cargo install` by default does not use the lockfile.

- 🤏 This template is a minimal (in terms of complexity and the number of components)

> template for building a blockchain node.

Reviewer: Resolving this since the same subject is discussed under a more actual comment.

Please see installation section on [crates.io/omni-node](https://crates.io/crates/polkadot-omni-node).

> #### Build `minimal-template-runtime` (A reminder to  build on the most updated version of polkadot-sdk master)

Reviewer: We can remove the reminder addition. We still waant people to refer to the dedicated minimal-template repository, so building out of `polkadot-sdk` can be done, but isn't the user flow we're looking. This readme will end up in this minimal-template dedicated repo, so we shouldn't refer back to `polkadot-sdk` from there.

```sh
cargo build -p minimal-template-runtime --release
```

#### Install `staging-chain-spec-builder`
  ```bash
  cargo install --path substrate/bin/utils/chain-spec-builder
```

> Or Please see the installation section at [`crates.io/staging-chain-spec-builder`](https://crates.io/crates/staging-chain-spec-builder).

Reviewer: Better to revert this because we will assume this readme lives in https://github.com/paritytech/polkadot-sdk-minimal-template.

We can install `zombienet` as described [here](https://paritytech.github.io/zombienet/install.html#installation),
and `zombienet-omni-node.toml` contains the network specification we want to start.

> To simplify the process of starting the minimal template with zombienet and Omni Node, we've added a pre-configured 
> development chain spec (dev_chain_spec.json) to the minimal template. The zombienet-omni-node.toml file of this template
> points to it, but you can update it to an updated path to the new chain spec generated on your machine, as shown below.

Reviewer: ```suggestion
To simplify the process of starting the minimal template with zombienet and Omni Node, we've added a pre-configured 
development chain spec (dev_chain_spec.json) to the minimal template. The zombienet-omni-node.toml file of this template
points to it, but you can update it to an updated path to the new chain spec generated on your machine.
```

My initial suggestion was to move this right into the section: `Update zombienet-omni-node.toml with a valid chain spec path`. It can be placed right at the beginning of the section, but then I would change the next paragraph: `Before starting...` by removing the first sentence.

Author: I made the changes to this
